lib/internal/http.js

```diff
@@ -112,7 +112,7 @@ class ProxyConfig {
     const { host, hostname, port, protocol, username, password } = new URL(proxyUrl);
     this.href = proxyUrl; // Full URL of the proxy server.
     this.host = host; // Full host including port, e.g. 'localhost:8080'.
-    this.hostname = hostname.replace(/^\[|\]$/g, ''); // Trim off the brackets from IPv6 addresses.
+    this.hostname = hostname[0] === '[' ? hostname.slice(1, -1) : hostname; // Trim off the brackets from IPv6 addresses.
```

Comment:
This change has the assumption the hostname always has the last bracket in the end if it starts with brackets

@anonrig since you are more familiar with urls, is this always true?

Also, you need to change this to prototype slice

Reply:
Yes. This is correct as per WHATWG URL. If the host string begins with [, it must end ] if it is to be valid.

Reply:
😎

Reply:
@Uzlopak Yeah, since the other part of the code is using StringPrototype, this should use aswell.

Thanks for the knowledge Lemire